hca/added_command.py

```diff
@@ -207,27 +212,103 @@ def _build_body_payload(cls, namespace):
             if arg_name not in endpoint_info["options"]:
                 continue
 
-            payload_format = endpoint_info["options"][arg_name]["in"]
+            payload_format = endpoint_info['options'][arg_name].get('in', None)
             if payload_format == "body":
                 hierarchy = endpoint_info["options"][arg_name]["hierarchy"]
                 cls._add_arg(arg, body_payload, hierarchy)
         return body_payload
 
     @classmethod
-    def _get_auth_header(cls, real_header=True):
-        try:
-            credentials, project_id = auth.default(scopes=["https://www.googleapis.com/auth/userinfo.email"])
-
-            r = Request()
-            credentials.refresh(r)
-            r.session.close()
+    def _get_auth_header(cls, args, retry=False):
+        """
+        Called in run when a user calls an authenticated method. retry=True if authenticated method returns 401.
 
-            token = credentials.token if real_header else str(uuid.uuid4())
+        :param args: Dict of arguments for any given function.
+        :param retry: Boolean indicating if this is the second time trying to authenticate. If so, refresh token.
+        """
+        token = cls._get_access_token(args, retry)
+        return {'Authorization': "Bearer {}".format(token)}
 
-            return "Bearer {}".format(token)
+    @classmethod
+    def _get_access_token(cls, args, retry):
+        config = Config(Constants.TWEAK_PROJECT_NAME)
+        access_token = None
+
+        # kwargs access_token input
+        if 'access_token' in args:
+            if retry:
+                logging.info("Access token taken from kwargs invalid.")
+                raise ValueError("The access token you've supplied in the kwargs is not valid."
+                                 " Please refresh and try again. You may also run `hca login`"
+                                 " to get a more permanent hca configuration.")
+            else:
+                logging.info("Found access token in kwargs.")
+                access_token = args['access_token']
+
+        # Checking config
+        elif config.get('access_token', None):
```

Comment:
In the test contextmanager, I tried to reset config by using something along the lines of `del config['access_token']`, but that didn't actually remove that variable from the config. So instead, I set them to `None` after I was done using them. This is a result of that. The only other solution I see around this is changing Tweak. @kislyuk am I misusing this?